File: R/2103-rings-and-rods/main.go
package main

import "fmt"

func countPoints(rings string) int {
	buf := make([][]int, 10)
	for i := 0; i < 10; i++ {
		buf[i] = make([]int, 3)
	}
	for i := 0; i < len(rings); i = i + 2 {
		idx := -1
		if byte(rings[i]) == byte('R') {
			idx = 0
		} else if byte(rings[i]) == byte('G') {
			idx = 1
		} else if byte(rings[i]) == byte('B') {
			idx = 2
		}
		offset := int(byte(rings[i+1]) - byte('0'))
		buf[offset][idx]++
	}
	cnt := 0
	for i := 0; i < 10; i++ {
		if buf[i][0] > 0 && buf[i][1] > 0 && buf[i][2] > 0 {
			cnt++
		}
	}
	return cnt
}

func main() {
	fmt.Println()
}

File: R/2103-rings-and-rods/main.rs
struct Solution {}

impl Solution {
  pub fn count_points(rings: String) -> i32 {
    let mut buf = vec![vec![0; 3]; 10];
    let bb = rings.as_bytes();
    (0..bb.len()).step_by(2).for_each(|idx| {
      let idx1 = match bb[idx] {
        b'R' => 0,
        b'G' => 1,
        _ => 2,
      };
      let offset = (bb[idx + 1] - b'0') as usize;
      buf[offset][idx1] += 1;
    });

    buf
      .iter()
      .filter(|b| b[0] > 0 && b[1] > 0 && b[2] > 0)
      .count() as i32
  }
}
